client/asset/eth/multirpc.go

```diff
@@ -1599,13 +1606,71 @@ func newCompatibilityTests(cb bind.ContractBackend, net dex.Network, log dex.Log
 	}
 }
 
-func domain(host string) string {
-	parts := strings.Split(host, ".")
-	n := len(parts)
-	if n <= 2 {
-		return host
+// domain accepts an url, ip, or file path and returns the domain:port if they
+// exist. Returns just the domain if no port. Returns a cleaned file path if a
+// file with .ipc suffix, otherwise returns the address as is if no errors were
+// encountered.
+func domain(addr string) (string, error) {
```

Comment:
Looks like we can plug it back into `failedProviders` now, if we like.  Reversing https://github.com/decred/dcrdex/pull/2273/files